### PR TITLE
Fix GHA cache key for go modules

### PR DESCRIPTION
**Pull request by spring1843**
…o.sum as key

**1. Issue, if available:**

**2. Description of changes:**
Attempt to make the GHA for CI faster by properly caching the go modules and build. Currently it seems like the cache is not as effective since the dependencies are still being deployed 

**3. How was this change tested?**

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
           ~/go/pkg/mod
           ~/go/bin/
           ~/.kubebuilder/bin/k8s
-        key: gocache
+        key: go-cache-${{ hashFiles('**/go.sum') }}
```
Comment (bwagner5) [inline .github/workflows/ci.yaml:23]: how does this make it faster?  Wouldn't the `gocache` get the same amount of hits (or maybe more) than hashing the deps checksums?

**Comment by spring1843** [comment]
I tried a few different things here it doesn't look like the only improvement we can make and hope for the best is changing this key, and it will become _better_.

The problem is in very few cases there is a hit to the cache, in the other cases when it tries to save the cache it fails with this error: `It doesn't look like there are problems with the caching settings`.  The recommendation [here](https://github.com/actions/toolkit/issues/658#issuecomment-1019398065) and [here](https://github.com/actions/toolkit/issues/658#issuecomment-1019398065) is to change the key.

This will cause the cache creation to happen only when there's a real change in dependencies, and the unique key helps in not trying to overwrite an existing cache. This key is recommended [here](https://markphelps.me/posts/speed-up-your-go-builds-with-actions-cache/) and I've seen it work for another Go project so I believe replacing this key will give us more consistent results.

**Comment by ellistarn** [comment]
> By "each suite" do you mean each of the tests we run for different kubernetes versions? or something else?

Each suite_test.go in our packages stands up a new kubernetes API server and ETCD, which takes about 10 seconds. IIRC ginkgo doesn't have a `BeforeAll` `AfterAll` directive, but `testenv` does allow you to reattach to an existing cluster. This can totally be a separate change though.

**Review by bwagner5 — APPROVED** [review]
lgtm, we can follow-up w/ more improvements 🚀

**Comment by spring1843** [comment]
>  IIRC ginkgo doesn't have a BeforeAll AfterAll directive, but testenv does allow you to reattach to an existing cluster. This can totally be a separate change though.

It looks like the Ginkgo v2 DSL supports BeforeAll and AfterAll: https://github.com/onsi/ginkgo/blame/master/dsl/core/core_dsl.go#L56

**Comment by ellistarn** [comment]
Super cool!

**Pull request merged**